incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaProxyFactory.java

```diff
@@ -71,13 +71,19 @@ public void attach(
     {
         MqttKafkaBindingConfig kafkaBinding = new MqttKafkaBindingConfig(binding);
         bindings.put(binding.id, kafkaBinding);
+
+        MqttKafkaSessionFactory sessionFactory = (MqttKafkaSessionFactory) factories.get(MqttBeginExFW.KIND_SESSION);
+        sessionFactory.onAttached(binding.id);
     }
 
     @Override
     public void detach(
         long bindingId)
     {
         bindings.remove(bindingId);
+
+        MqttKafkaSessionFactory sessionFactory = (MqttKafkaSessionFactory) factories.get(MqttBeginExFW.KIND_SESSION);
+        sessionFactory.onDetached(bindingId);
```

Comment:
Create an interface called `MqttKafkaStreamFactory` with `default` empty implementations of `onAttached` and `onDetached`, then make each factory implement the interface, overriding `onAttached` and `onDetached` on session factory.

Update `factories` map to have target type of `MqttKafkaStreamFactory` and eliminate cast here in `detach` and above in `attach` too.

Instead of special casing session factory, you could just call `onAttached` / `onDetached` on all the factories, some of them will not react as they just have the default implementation, but session factory will have some session-specific behavior.

```
factories.values().forEach(MqttStreamFactory::onAttached);
```
and
```
factories.values().forEach(MqttStreamFactory::onDetached);
```